# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/Scope.cpp
# Commit: 11fc8f557a2a
# Full Hash: 11fc8f557a2a7cf3c89e7c1396316aabd9725778
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-03-06 09:33:16
# Regressor Bug: 1689413
# File Overlap Count: 1
# Description:
#   Bug 1689413 part 6 - Move Realm* from ObjectGroup to BaseShape. r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D106976
# ==============================================================================

diff -r 9bfa6023a553 -r 11fc8f557a2a js/src/vm/Scope.cpp
--- a/js/src/vm/Scope.cpp	Sat Mar 06 01:05:20 2021 +0000
+++ b/js/src/vm/Scope.cpp	Sat Mar 06 01:05:21 2021 +0000
@@ -96,8 +96,8 @@
                                  uint32_t numSlots, ObjectFlags objectFlags) {
   // Put as many slots into the object header as possible.
   uint32_t numFixed = gc::GetGCKindSlots(gc::GetGCObjectKind(numSlots));
-  return EmptyShape::getInitialShape(cx, cls, TaggedProto(nullptr), numFixed,
-                                     objectFlags);
+  return EmptyShape::getInitialShape(cx, cls, cx->realm(), TaggedProto(nullptr),
+                                     numFixed, objectFlags);
 }
 
 static Shape* NextEnvironmentShape(JSContext* cx, HandleAtom name,
@@ -542,9 +542,9 @@
 }
 
 Shape* Scope::maybeCloneEnvironmentShape(JSContext* cx) {
-  // Clone the environment shape if cloning into a different zone.
+  // Clone the environment shape if cloning into a different realm.
   Shape* shape = environmentShape();
-  if (shape && shape->zoneFromAnyThread() != cx->zone()) {
+  if (shape && shape->realm() != cx->realm()) {
     BindingIter bi(this);
     return CreateEnvironmentShape(cx, bi, shape->getObjectClass(),
                                   shape->slotSpan(), shape->objectFlags());